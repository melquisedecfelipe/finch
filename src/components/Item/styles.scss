.dashboard-item {
  background: #FFFFFF;
  box-shadow: 0px 10px 20px #16319B29;
  border-radius: 3px;

  & .item-img {
    position: relative;
    width: 100%;
    height: 180px;
    background: no-repeat;
    background-size: cover;
    background-position: center center;

    > p {
      position: absolute;
      bottom: 0;
      right: 0;
      font-size: 14px;
      line-height: 21px;
      color: #5E3BA5;
      color: #FFFFFF;
      padding: 5px 20px;
    }

    > .-red {
      background: transparent linear-gradient(125deg, #EF0E5A 0%, #55118F 100%);
    }

    > .-blue {
      background: transparent linear-gradient(125deg, #1DB6CE 0%, #182AC9 100%);
    }
  }

  > .item-content {
    padding: 20px;

    > .item-header {
      display: flex;
      justify-content: space-between;
      align-items: center;

      > h1 {
        font-weight: bold;
        font-size: 20px;
        line-height: 33px;
        color: #5E3BA5;
      }

      > .item-check {
        display: flex;
        justify-content: center;
        align-items: center;

        > label {
          position: relative;
          display: inline-block;
          width: 26px;
          height: 15px;

          > input {
            opacity: 0;
            width: 0;
            height: 0;
          }

          > span {
            position: absolute;
            cursor: pointer;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            background-color: #ccc;
            -webkit-transition: .4s;
            transition: .4s;
            border-radius: 34px;

            &:before {
              position: absolute;
              content: "";
              height: 10px;
              width: 10px;
              left: 3px;
              bottom: 3px;
              background-color: white;
              -webkit-transition: .4s;
              transition: .4s;
              border-radius: 50%;
            }
          }

          input:checked + span {
            background-color: rgb(0, 134, 248);
          }

          input:checked + span:before {
            -webkit-transform: translateX(10px);
            -ms-transform: translateX(10px);
            transform: translateX(10px);
          }

        }
        p {
          font-size: 12px;
          line-height: 21px;
          color: #8E879E;
          margin-left: 5px;
        }
      }
    }

    > h1 {
      font-weight: bold;
      font-size: 20px;
      line-height: 33px;
      color: #282430;
      margin-top: 10px;
    }

    > p {
      font-size: 14px;
      line-height: 23px;
      color: #8E879E;
    }
  }
}
