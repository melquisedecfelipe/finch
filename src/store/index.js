import { createStore } from 'redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from './ducks';

const store = createStore(rootReducer, composeWithDevTools());

export default store;
