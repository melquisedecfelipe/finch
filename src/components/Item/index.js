import React from 'react';
import { Link } from 'react-router-dom';

import './styles.scss';

export default function Dashboard({ item }) {
  const { id, exclusivo, promocao, valor, favoritos, nome, decricaoCurta } = item;

  function handleCheckbox() {
    const localStorageItems = JSON.parse(localStorage.getItem('items'));

    const newFavorite = localStorageItems.map(elem => {
      return elem.id === id ? { ...elem, favoritos: !elem.favoritos } : elem;
    });

    localStorage.setItem('items', JSON.stringify(newFavorite));
  }

  return (
    <div className="dashboard-item">
      <Link to={`/item/${id}`}>
        <div className="item-img" style={{ backgroundImage: `url(${item.imagem})` }}>
          <p className={exclusivo === true ? '-blue' : '' || promocao === true ? '-red' : ''}>
            {exclusivo === true ? 'Exclusivo' : '' || promocao === true ? 'Promoção' : ''}
          </p>
        </div>
      </Link>
      <div className="item-content">
        <div className="item-header">
          <h1>
            {parseInt(valor, 10).toLocaleString('pt-br', {
              style: 'currency',
              currency: 'BRL',
            })}
          </h1>
          <div className="item-check">
            <label htmlFor="checkbox">
              <input
                type="checkbox"
                id="checkbox"
                value={favoritos}
                onChange={() => handleCheckbox()}
                defaultChecked={favoritos === true}
              />
              <span />
            </label>
            <p>tornar favorito</p>
          </div>
        </div>
        <h1>{nome}</h1>
        <p>{decricaoCurta}</p>
      </div>
    </div>
  );
}
