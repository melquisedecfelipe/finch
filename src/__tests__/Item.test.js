import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Item from '../components/Item';

import store from '../store';

describe('Item page testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Item />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
