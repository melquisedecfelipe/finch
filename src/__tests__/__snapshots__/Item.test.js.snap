// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item page testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
